src/uct/cuda/base/cuda_md.c

```diff
@@ -74,7 +74,7 @@ uct_cuda_base_query_md_resources(uct_component_t *component,
     char device_name[10];
     int num_gpus;
 
-    cudaErr = cudaGetDeviceCount(&num_gpus);
+    cudaErr = UCS_PROFILE_CALL_ALWAYS(cudaGetDeviceCount, &num_gpus);
     if ((cudaErr != cudaSuccess) || (num_gpus == 0)) {
```

Comment:
I think we should print error here in case of failed CUDA call.
```suggestion
     status = UCT_CUDA_CALL_LOG_ERR(cudaGetDeviceCount, &num_gpus);
    if ((status != UCS_OK) || (num_gpus == 0)) {
```

Reply:
maybe ucs_diag?

Reply:
This is usually the first CUDA Runtime call from UCX.
If the call returns error, then the query will return empty MD, and CUDA transports will not be supported.
We should print an error here, IMO.

Reply:
So far the policy is to not print errors if something is not supported.
Do you know in which cases cudaGetDeviceCount would return error?
@Akshay-Venkatesh WDYT?

Reply:
@rakhmets fixed (force pushed due to small pr)